src/main/java/de/tum/in/www1/artemis/domain/GradeStep.java

```diff
@@ -0,0 +1,148 @@
+        return lowerBoundPercentage;
+    }
+
+    public void setLowerBoundPercentage(double lowerBoundPercentage) {
+        this.lowerBoundPercentage = lowerBoundPercentage;
+    }
+
+    public boolean isLowerBoundInclusive() {
+        return lowerBoundInclusive;
+    }
+
+    public void setLowerBoundInclusive(boolean lowerBoundInclusive) {
+        this.lowerBoundInclusive = lowerBoundInclusive;
+    }
+
+    public double getUpperBoundPercentage() {
+        return upperBoundPercentage;
+    }
+
+    public void setUpperBoundPercentage(double upperBoundPercentage) {
+        this.upperBoundPercentage = upperBoundPercentage;
+    }
+
+    public boolean isUpperBoundInclusive() {
+        return upperBoundInclusive;
+    }
+
+    public void setUpperBoundInclusive(boolean upperBoundInclusive) {
+        this.upperBoundInclusive = upperBoundInclusive;
+    }
+
+    public String getGradeName() {
+        return gradeName;
+    }
+
+    public void setGradeName(String gradeName) {
+        this.gradeName = gradeName;
+    }
+
+    public boolean isPassingGrade() {
+        return isPassingGrade;
+    }
+
+    public void setPassingGrade(boolean passingGrade) {
+        isPassingGrade = passingGrade;
+    }
+
+    @Override
+    public String toString() {
+        return "GradeStep{" + "lowerBoundPercentage=" + lowerBoundPercentage + ", lowerBoundInclusive=" + lowerBoundInclusive + ", upperBoundPercentage=" + upperBoundPercentage
+                + ", upperBoundInclusive=" + upperBoundInclusive + ", gradeName='" + gradeName + '\'' + ", isPassingGrade=" + isPassingGrade + '}';
+    }
+
+    /**
+     * Returns whether the given percentage maps to this grade step
+     * - the percentage should be between the lower and upper bounds of the grade step
+     * - or on the bound of it if it's inclusive
+     *
+     * @param percentage the percentage which has to be mapped
+     * @return whether the percentage matches this grade step
+     */
+    public boolean matchingGradePercentage(double percentage) {
+        if (percentage == lowerBoundPercentage) {
+            return lowerBoundInclusive;
+        }
+        else if (percentage == upperBoundPercentage) {
+            return upperBoundInclusive;
+        }
+        else {
+            return percentage > lowerBoundPercentage && percentage < upperBoundPercentage;
+        }
+    }
+
+    /**
+     * Checks if the current grade step is valid
+     * - the grade name should be set and it shouldn't be empty
+     * - both bounds should be between 0 and 100 (both inclusive)
+     * - the lower bound should be strictly less than the upper bound
+     *
+     * @return true if all conditions are true and false otherwise
+     */
+    public boolean isValid() {
+        return gradeName != null && !gradeName.equals("") && lowerBoundPercentage >= 0 && upperBoundPercentage > 0 && lowerBoundPercentage < upperBoundPercentage
+                && lowerBoundPercentage < 100 && upperBoundPercentage <= 100;
```

Comment:
Fwiw, `upperBoundPercentage > 0` and `lowerBoundPercentage < 100` isn't needed because of the `lowerBoundPercentage < upperBoundPercentage` check, but it doesn't hurt either so I won't insist on removing those.

Reply:
Will change it in the next commit